rasa/cli/export.py

```diff
@@ -0,0 +1,508 @@
+        tracker_store: Tracker store to source events from.
+        minimum_timestamp: Minimum timestamp of events that are published. If `None`,
+            apply no such constraint.
+        maximum_timestamp: Maximum timestamp of events that are published. If `None`,
+            apply no such constraint.
+         conversation_ids: List of conversation IDs selected for publishing.
+
+    Returns:
+        List of serialized events with added `sender_id` field.
+
+    """
+    cli_utils.print_info(
+        f"Fetching events for {len(conversation_ids)} conversation IDs:"
+    )
+
+    events = []
+
+    for conversation_id in tqdm(conversation_ids, "conversation IDs"):
+        tracker = tracker_store.retrieve(conversation_id)
+        if not tracker:
+            logger.info(
+                f"Could not retrieve tracker for conversation ID "
+                f"'{conversation_id}'. Skipping."
+            )
+            continue
+
+        _events = tracker.current_state(EventVerbosity.ALL)["events"]
+
+        if not _events:
+            logger.info(
+                f"No events to migrate for conversation ID '{conversation_id}'."
+            )
+
+        # the conversation IDs are needed in the event publishing
+        for event in _events:
+            event["sender_id"] = conversation_id
+            events.append(event)
+
+    return _sort_and_select_events_by_timestamp(
+        events, minimum_timestamp, maximum_timestamp
+    )
+
+
+def _get_continuation_command(
+    timestamp: float,
+    maximum_timestamp: Optional[float] = None,
+    endpoints_path: Optional[Text] = None,
+    requested_conversation_ids: Optional[List[Text]] = None,
+) -> Text:
+    """Build CLI command to continue 'rasa export' where it was interrupted.
+
+    Called when event publishing stops due to an error.
+
+    Args:
+        timestamp: Timestamp of the last event attempted to be published.
+        maximum_timestamp: Maximum timestamp of events that are published. If `None`,
+            apply no such constraint.
+        endpoints_path: Path to the endpoints file used to configure the event broker
+            and tracker store. If `None`, the default path ('endpoints.yml') is used.
+        requested_conversation_ids: List of conversation IDs that should be published
+            requested by the user. If `None`, all conversation IDs contained in the
+            tracker store are published.
+
+    """
+    # build CLI command command based on supplied timestamp and options
+    command = f"rasa export"
+
+    if endpoints_path is not None:
+        command += f" --endpoints {endpoints_path}"
+
+    command += f" --minimum-timestamp {timestamp}"
+
+    if maximum_timestamp is not None:
+        command += f" --maximum-timestamp {maximum_timestamp}"
+
+    if requested_conversation_ids:
+        command += f" --conversation-ids {','.join(requested_conversation_ids)}"
+
+    return command
+
+
+def _publish_events(
+    tracker_store: TrackerStore,
```

Comment:
optional: this is a long parameter list which is passed around. Consider doing a more object oriented approach and have a class `Migrator` which can make use of it's attributes.

Reply:
I would agree in principle, but these parameters are only passed to `_publish_events()` - there's no need to abstract into a class yet

Reply:
But in `_publish` you're passing things to `__fetch_events_within_time_range` which in turn passes things to `_sort_and_select_events_by_timestamp` . So there is actually quite a lot of passing things around.

Reply:
added a migrator class